sdks/python/apache_beam/io/gcp/bigquery_read_internal.py

```diff
@@ -100,3 +127,258 @@ def process(self, unused_element, unused_signal, gcs_locations):
+            self.bigquery_job_labels))
+    job_ref = job.jobReference
+    bq.wait_for_bq_job(job_ref, max_retries=0)
+    return bq._get_temp_table(self._get_project())
+
+  def _export_files(
+      self,
+      bq: bigquery_tools.BigQueryWrapper,
+      element: 'ReadFromBigQueryRequest',
+      table_reference: TableReference):
+    """Runs a BigQuery export job.
+
+    Returns:
+      bigquery.TableSchema instance, a list of FileMetadata instances
+    """
+    job_labels = self._get_bq_metadata().add_additional_bq_job_labels(
+        self.bigquery_job_labels)
+    export_job_name = bigquery_tools.generate_bq_job_name(
+        self._job_name,
+        self._source_uuid,
+        bigquery_tools.BigQueryJobTypes.EXPORT,
+        element.obj_id)
+    temp_location = self.options.view_as(GoogleCloudOptions).temp_location
+    gcs_location = bigquery_export_destination_uri(
+        self.gcs_location,
+        temp_location,
+        '%s%s' % (self._source_uuid, element.obj_id))
+    if self.use_json_exports:
+      job_ref = bq.perform_extract_job([gcs_location],
+                                       export_job_name,
+                                       table_reference,
+                                       bigquery_tools.FileFormat.JSON,
+                                       project=self._get_project(),
+                                       job_labels=job_labels,
+                                       include_header=False)
+    else:
+      job_ref = bq.perform_extract_job([gcs_location],
+                                       export_job_name,
+                                       table_reference,
+                                       bigquery_tools.FileFormat.AVRO,
+                                       project=self._get_project(),
+                                       include_header=False,
+                                       job_labels=job_labels,
+                                       use_avro_logical_types=True)
+    bq.wait_for_bq_job(job_ref)
+    metadata_list = FileSystems.match([gcs_location])[0].metadata_list
+
+    if isinstance(table_reference, ValueProvider):
+      table_ref = bigquery_tools.parse_table_reference(
+          element.table, project=self._get_project())
+    else:
+      table_ref = table_reference
+    table = bq.get_table(
+        table_ref.projectId, table_ref.datasetId, table_ref.tableId)
+
+    return table.schema, metadata_list
+
+  def _get_project(self):
+    """Returns the project that queries and exports will be billed to."""
+
+    project = self.options.view_as(GoogleCloudOptions).project
+    if isinstance(project, ValueProvider):
+      project = project.get()
+    if not project:
+      project = self.project
+    return project
+
+
+FieldSchema = collections.namedtuple('FieldSchema', 'fields mode name type')
+
+
+class _JsonToDictCoder(coders.Coder):
+  """A coder for a JSON string to a Python dict."""
+  def __init__(self, table_schema):
```

Comment:
Please add unit tests for the new coder.

Reply:
The coder was just moved from bq.py to bq_read_internal.py